### ISSUE TITLE
Code Correctness: Function Not Invoked

### ISSUE THREAD

**Issue by gopalchinnadurai** [SDWebImage/SDWebImage#2911]
We have used **SDWebImage 5.4.0** pod file as per our need. Some VAPT points are mentioned in below picture. Kindly help us asap.

**Issue Summary:**
Because it is missing trailing parentheses, the expression on line **66** in **UIImage+MultiFormat.m** refers to the value of a function pointer rather than the return value of the function.Because it is missing trailing parentheses, this expression refers to the value of the function pointer rather than the return value of the function.

**Issue Screenshot:**

<img width="534" alt="Screenshot 2019-12-23 at 11 31 02 AM" src="https://user-images.githubusercontent.com/55129685/71339408-6db3ec80-2579-11ea-9fec-5f026f0cd4a8.png">

**Comment by dreampiggy** [comment]
> The expression on line 66 in UIImage+MultiFormat.m

Are you sure you're running this CVE report on SDWebImage 5.4.0 ? There is no 66 line at all.

Check this: https://github.com/SDWebImage/SDWebImage/blob/5.4.0/SDWebImage/Core/UIImage+MultiFormat.m

And it does not use `Function Pointer` at all, it use Objective-C method, not C function. Using the return value as it is.

**Comment by dreampiggy** [comment]
I guess you mean 4.4.0:
https://github.com/SDWebImage/SDWebImage/blob/4.4.0/SDWebImage/UIImage+MultiFormat.m

There is one code :

```objective-c
    if (self) { // line 66
        imageData = [[SDWebImageCodersManager sharedInstance] encodedDataWithImage:self format:imageFormat];
    }
```

This if check can be removed since it's always valid.

**Comment by gopalchinnadurai** [comment]
Following Screenshot, my library having:

**Line No: 66**
============================================================

<img width="833" alt="Screenshot 2019-12-23 at 12 31 25 PM" src="https://user-images.githubusercontent.com/55129685/71341585-46145280-2580-11ea-9c1b-8a1e98c8e143.png">

**Comment by dreampiggy** [comment]
@opalchinnadurai  You're using the 4.4.7 version. Actually

This code is designed behavior, it's used to check `CGImageGetUTType` this function is available at runtime. Not calling it.

And sadly, we can not using `@available (iOS 9.0, *)` this code, because 4.x should support Xcode 8. Which does not have the available runtime check.

I think you should just ignore that static analysis tool's report.

**Comment by gopalchinnadurai** [comment]
Thanks for your support. @dreampiggy
